doc.md

```diff
@@ -0,0 +1,34 @@
+`PartialOrd` Documentation
+==========================
+
+[Strict partial ordering relation](https://en.wikipedia.org/wiki/Partially_ordered_set#Strict_and_non-strict_partial_orders).
```

Comment:
In general: I think the first two paragraph are a bit too technical for most users. I think it would be a bit better if we first describe the trait in easy terms for the common user. For example, add something like this in the beginning:

> In Rust, there are actually two traits related to ordering/comparisons: `PartialOrd` and `Ord` where the former is a super trait of the latter. This stems from the fact that many sets contain elements that can be compared, but these comparisons don't necessarily satisfy some nice-to-have properties. The easiest example are integers, like `i32`. Integers have these nice properties, in particular every integer is comparable to every other integer.
>
> However, there are plenty of ordering relations with elements that are not comparable. For example, [the "subset relation" on sets](https://en.wikipedia.org/wiki/Subset) in which, for sets `a` and `b`, `a < b` is true if and only if `a` is a proper subset of `b` (e.g. `{4, 9} < {1, 4, 5, 9}` is true). The problem is that when neither `a` is a subset of `b` nor `b` is a subset of `a` nor `a = b` (e.g. `{1, 3}` and `{1, 4}`), we cannot sensible compare `a` and `b`. 
>
> Another popular example are the IEEE-754 floating point numbers. There exist the special `NaN` value which cannot be compared to any other value. 
>
> The `PartialOrd` has fewer requirements of the ordering relation than `Ord`. In particular, `partial_cmp` can return `None` if two elements cannot be compared. That means that `PartialOrd` can be implemented for floats and for ordering relations like subset relations.

(this is just a rough sketch to illustrate what kind of information and in what kind of formulation I mean)

Reply:
> Maybe something like this?
>
>> Defines the operators <, >, <= and >= for comparing values, where < and > form strict 
>> partial ordering relations.

I think it would be worth it to re-state here that these operators always return `bool`.